tests/test_opencode_security_boundaries.py

```
@@ -134,6 +325,289 @@ def test_sensitive_log_redaction_handles_lists_empty_input_and_cli(monkeypatch:
+    opaque_secret = "-".join(("opaque", "fixture", "secret", "123456"))
+    provider_body = "B" * 24
+    source = (
+        "ordinary build failure\n"
+        f"to\x1b[31mken\x1b[0m={opaque_secret}\n"
+        f"to\x1b(Bken={opaque_secret}\n"
+        f"to\x1b7ken={opaque_secret}\n"
+        f"to\x9b31mken={opaque_secret}\n"
+        f"provider ghp_\x1b[32m{provider_body}\x1b[0m\n"
+        + json.dumps(
+            {
+                f"to\x1b[31mken": opaque_secret,
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)